### PR TITLE
Fix InfinispanTest failed on Mac #2840

### PR DESCRIPTION
**Pull request by ffang**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: integration-tests/infinispan/src/test/java/org/apache/camel/quarkus/component/infinispan/InfinispanServerTestResource.java [hunk 1]
```diff
@@ -54,24 +55,48 @@
...
+                        "camel.component.infinispan.autowired-enabled", "false",
+                        "camel.component.infinispan.hosts", serverList,
+                        "camel.component.infinispan.username", USER,
+                        "camel.component.infinispan.password", PASS,
+                        "camel.component.infinispan.secure", "true",
+                        "camel.component.infinispan.security-realm", "default",
+                        "camel.component.infinispan.sasl-mechanism", "DIGEST-MD5",
+                        "camel.component.infinispan.security-server-name",
+                        "infinispan");
+            }
```
Comment (ppalaga) [inline integration-tests/infinispan/src/test/java/org/apache/camel/quarkus/component/infinispan/InfinispanServerTestResource.java:99]: Could we perhaps reduce the number of lines a bit? Maybe something like the following?

```suggestion
            Map<String, String> result = CollectionHelper.mapOf(
                        // quarkus
                        "quarkus.infinispan-client.server-list", serverList,
                        "quarkus.infinispan-client.near-cache-max-entries", "3",
                        "quarkus.infinispan-client.auth-username", USER,
                        "quarkus.infinispan-client.auth-password", PASS,
                        "quarkus.infinispan-client.auth-realm", "default",
                        "quarkus.infinispan-client.sasl-mechanism", "DIGEST-MD5",
                        "quarkus.infinispan-client.auth-server-name", "infinispan",
                        // camel
                        "camel.component.infinispan.autowired-enabled", "false",
                        "camel.component.infinispan.hosts", serverList,
                        "camel.component.infinispan.username", USER,
                        "camel.component.infinispan.password", PASS,
                        "camel.component.infinispan.secure", "true",
                        "camel.component.infinispan.security-realm", "default",
                        "camel.component.infinispan.sasl-mechanism", "DIGEST-MD5",
                        "camel.component.infinispan.security-server-name",
                        "infinispan");
            if (SystemUtils.IS_OS_MAC) {
                /* Fix for https://github.com/apache/camel-quarkus/issues/2840 */
                result.put("quarkus.infinispan-client.client-intelligence", "BASIC");
            }
            return result;
```

**Comment by zbendhiba** [comment]
I prefer the `SystemUtils.IS_OS_MAC` as suggested by @ppalaga.
Because, we don't know if the `result.put("quarkus.infinispan-client.client-intelligence", "BASIC");` is valid for all non OS_LINUX.

File: integration-tests/infinispan/src/test/java/org/apache/camel/quarkus/component/infinispan/InfinispanServerTestResource.java (hunk 1 above)
Comment (ffang) [reply integration-tests/infinispan/src/test/java/org/apache/camel/quarkus/component/infinispan/InfinispanServerTestResource.java:99]: Hi @ppalaga ,

Thanks for the feedback and I will go this way.
I think we need to add this property for windows machine as well. The root cause is same for windows and mac. The Docker desktop can't route request to internal docker IP directly on mac and  windows.

Please see this discussion
https://stackoverflow.com/questions/63465221/running-infinispan-in-docker-on-windows-10
Freeman

**Comment by ffang** [comment]
> I prefer the `SystemUtils.IS_OS_MAC` as suggested by @ppalaga.
> Because, we don't know if the `result.put("quarkus.infinispan-client.client-intelligence", "BASIC");` is valid for all non OS_LINUX.

Hi @zbendhiba ,

At least for windows we also need this property, please see this discussion
https://stackoverflow.com/questions/63465221/running-infinispan-in-docker-on-windows-10

Cheers
Freeman

File: integration-tests/infinispan/src/test/java/org/apache/camel/quarkus/component/infinispan/InfinispanServerTestResource.java (hunk 1 above)
Comment (ppalaga) [reply integration-tests/infinispan/src/test/java/org/apache/camel/quarkus/component/infinispan/InfinispanServerTestResource.java:99]: Thanks, sounds good.

**Comment by ffang** [comment]
addressed feedback and squashed new commit https://github.com/apache/camel-quarkus/pull/2841/commits/1718479a0da7a7ac9bbf1d71502ad6aefcf1d35d

**Review by zbendhiba — APPROVED** [review]
@ffang thanks

**Review by jamesnetherton — APPROVED** [review]

**Comment by ffang** [comment]
Thanks All!

**Pull request merged**